Author: Niko Tyni <ntyni@debian.org>
Subject: Fix test failures when the build directory contains regexp metacharacters.
 The build directory name may contain regexp metacharacters like '+'
 that need escaping or they break the substitution.
Forwarded: no
Reviewed-by: Nicholas Bamber <nicholas@periapt.co.uk>
Last-Update: 2011-05-17
---
 lib/Devel/Cover.pm |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

--- a/lib/Devel/Cover.pm
+++ b/lib/Devel/Cover.pm
@@ -469,7 +469,7 @@
         # print STDERR "finally <$file> <$Dir>\n";
     }
     $file =~ s|\\|/|g if $^O eq "MSWin32";
-    $file =~ s|^$Dir/|| if defined $Dir;
+    $file =~ s|^\Q$Dir\E/|| if defined $Dir;
 
     $Digests ||= Devel::Cover::DB::Digests->new(db => $DB);
     $file = $Digests->canonical_file($file);
